src/NzbDrone.Core/Notifications/Trakt/TraktProxy.cs

```diff
@@ -0,0 +1,110 @@
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Notifications.Trakt.Resource;
+
+namespace NzbDrone.Core.Notifications.Trakt
+{
+    public interface ITraktProxy
+    {
+        string GetUserName(string accessToken);
+        HttpRequest GetOAuthRequest(string callbackUrl);
+        TraktAuthRefreshResource RefreshAuthToken(string refreshToken);
+        void AddToCollection(TraktCollectShowsResource payload, string accessToken);
+        HttpRequest BuildTraktRequest(string resource, HttpMethod method, string accessToken);
+    }
+
+    public class TraktProxy : ITraktProxy
+    {
+        private const string URL = "https://api.trakt.tv";
+        private const string OAuthUrl = "https://api.trakt.tv/oauth/authorize";
```

Comment:
Should be updated to remove the `api` subdomain: https://github.com/Sonarr/Sonarr/commit/6db24013592dddea5d9c0c4eff27b0da4071b646

Reply:
will look into this in the next few days.. sort of busy right now, but will look into this and the other comments soon.. :)